dashboard/app/models/levels/bubble_choice.rb

```diff
@@ -119,4 +128,11 @@ def best_result_sublevel(user)
     ul = user.user_levels.where(level: sublevels).max_by(&:best_result)
     ul&.level
   end
+
+  # Returns an array of BubbleChoice parent levels for any given sublevel name.
+  # @param [String] level_name. The name of the sublevel.
+  # @return [Array<BubbleChoice>] The BubbleChoice parent level(s) of the given sublevel.
+  def self.parent_levels(level_name)
+    where("properties -> '$.sublevels' LIKE ?", "%\"#{level_name}\"%")
```

Comment:
please confirm that this won't match a substring, eg searching for level_2018 will not match against a level with sublevel level_2018_2019.

a more robust way would be to use JSON_ARRAY to turn '$.sublevels' into an actual array, if possible.

Reply:
i have a unit test that tests that case and a similar one here: https://github.com/code-dot-org/code-dot-org/blob/004d6197acf582b2148a2a0b761269c45017bcee/dashboard/test/models/bubble_choice_test.rb#L182-L183